### DISCUSSION TITLE
Docs and romcal.js.org in romcal monorepo

### DISCUSSION THREAD

**Discussion by tukusejssirs** [romcal/romcal#537] [Ideas]
Here are my comments on the [description](https://github.com/romcal/romcal-examples/pull/7#issue-2270049303) of romcal/romcal-examples#7 written by @emagnier. As my comments are unrelated, I have created this discussion.

/cc @TobiTenno, @emagnier

---

> How do you think we should handle this? In the future, we might want to separate the example projects from the romcal.js.org site (to keep the example projects simple).

IMHO, we should create a separate package in the future monorepo of romcal just for romcal.js.org and publish a new version whenever that package is updated (which should also happen whenever some romcal library is updated) in GitHub CI.

> I'd like to work on a more comprehensive website, possibly using Docusaurus, including romcal documentation and allowing navigation somehow similar to what we have now

IMHO, the `docs` app should be separate from the calendar app (which is currently hosted at romcal.js.org). I think these apps have different scopes: one is used to document the usage, contribution, etc, while the other is show-casing how romcal can be used.

That said, `docs` could (and should) link to romcal.js.org, or even it could be published under romcal.js.org/docs, but IMHO it should still be a separate app and package.

And we might want to use Vitepress instead of Docusaurus, however, I don’t care much about it.

**Comment by TobiTenno** [comment]
I've already integrated the unique parts of #7 into my earlier pr, and it's just awaiting your eyes

**Reply by tukusejssirs** [reply]
Yeah, but this discussion is about something else. :wink:

And I am in the middle the review of romcal/romcal-examples#7.

**Comment by TobiTenno** [comment]
I'm against making another package specifically for romcal.js.org. there's no reason to publish it as a package or to provide it out from our infrastructure.

If we want to incorporate a standalone docs repo into romcal.js.org and shim the docs as a supage of romcal.js.org for docs, that makes sense, as romcal.js.org is essentially the simplest and most cohesive public-facing way for people to see what romcal is intending to provide.

Why do we need a docs "package"? Documentation isn't published or consumed by other projects. We have wikis that serve the purpose we'd need for markdown-formatted with code context & examples. Most of our docs pages would include the same content in probably about the same formatting.

**Reply by emagnier** [reply]
We agree that the documentation should remain as it is currently, versioned in markdown files in the main romcal repository (and therefore in one single place). It doesn't make sense to separate it into a separate project or package.

In the future, this documentation could potentially be automatically integrated into the romcal.js.org website (in addition to showcasing what romcal is intending to provide), to make it more accessible and pleasant to consult. That's all I was mentioning on my end (along with the idea of using Docusaurus, eventually, instead of the current example project, to keep these example projects as simple as possible, but we're not there yet).

> How do you think we should handle this?

My question here was just out of curiosity about the website update workflow and the feasibility of linking to another project to generate the romcal.js.org website, rather than the current React project. However, that's a question for the future. I'll wait to see what @TobiTenno will propose in the PR, so no need for further explanation here 👌.

**Reply by tukusejssirs** [reply]
### On the comment from @TobiTenno

> I'm against making another package specifically for romcal.js.org. there's no reason to publish it as a package or to provide it out from our infrastructure.

Well, as Etienne said, I too think that the example apps should be simple, however, the app running at romcal.js.org should be robust show-casing everything we can make using the romcal data. I consider the apps to be a guide for users how they can build their own apps.

In the future, IMHO romcal.js.org should include a sidebar/hamburger menu (or tabs or whatever else you like) with the following items:

- generated calendar for the selected calendar for the selected year in the selected locale (this is already done);
- calendar propers for each of the supported calendars (this is an unordered list);
- list of sources with references where the sources are cited;
- maybe even a kind of martyrology where all saints/blesseds and non-personal celebrations would be listed, along with all the metadata we have about them, probably including an image for each of them (the images would not be part of romcal);
- docs (from a separate internal app/package);
- unless we use external tools, we could also create tools (or customise existing ones) for:
   - source addition;
   - saints/blesseds or non-personal celebrations addition;
   - calendar addition;
   - celebration inscription/change in a calendar;
   - localisation of any string used in romcal.

Re publishing issue: sorry, I was not specific enough: neither I think the package for romcal.js should be published to NPM. When I say a _package_, I mean an app or a library in a monorepo which should not be necessarily _published to a registry_, but it could be an _internal package_ (app or library). Nx calls the non-buildable internal libraries [_workspace libraries_](https://nx.dev/concepts/more-concepts/buildable-and-publishable-libraries#publishable-libraries).

Anyway, I like the [Nx philosophy and mental model](https://nx.dev/concepts/more-concepts/applications-and-libraries#mental-model). It is what I (try to) use in my apps and libraries. YMMV.

> Why do we need a docs "package"? Documentation isn't published or consumed by other projects. We have wikis that serve the purpose we'd need for markdown-formatted with code context & examples. Most of our docs pages would include the same content in probably about the same formatting.

Same as above: `docs` should definitely be an _internal_ package, i.e. it should not be published to NPM.

Re wikis: If you mean GitHub Wikis, I am for moving the content of the one in the romcal repo to `docs`, as IMHO it should be versioned like romcal (a single version for `@romcal/data`, `@romcal/calendars`, `@romcal/core`, and `docs`; other packages could have different versioning or could be unversioned).

And basically, `docs` will be a package (specifically an app) if we use Vitepress/Docusaurus.

---

### On the comment from @emagnier

> We agree that the documentation should remain as it is currently, versioned in markdown files in the main romcal repository (and therefore in one single place). It doesn't make sense to separate it into a separate project or package.

I agree that for now, the docs could be simple MD files, however, I thought we all agree to create an app from them (e.g. using Vitepress or Docusaurus), then it is an app (thus package, albeit an internal/non-publishable one).

> along with the idea of using Docusaurus, eventually, instead of the current example project

Well, I thought that you were mentioning Docusaurus regarding the _documentation_ only, as it is usually used for the docs only.

**Reply by TobiTenno** [reply]
I agree that we _can_ use something like vitepress or docusaurus, but _why_ should we? what are the advantages over either github wikis or just markdown files in the repo?

obviously you addressed the clear benefit over github wikis that the source files can be versioned, but i believe that having another _separate_ docs workspace library or separate repository distances the docs even further from being equivalently versioned to the library itself than the github wikis do

**Reply by emagnier** [reply]
@tukusejssirs, you're diving into lots of ideas and projects. Maybe in the future, we'll see (if there are people willing to pitch in), but that's not the topic I initially wanted to discuss.

On my end, my original idea remains quite simple: to gather and present the documentation on romcal.js.org (in addition to the current calendar), in order to offer a better browsing experience, improved navigation through the documentation, search tools, SEO... essentially, to make the documentation even more accessible than just through the Github repo (it's a subject which is dear to me). Whatever technology is used for this (I probably shouldn't have mentioned Docusaurus at this point, as it adds unnecessary complexity to the discussion). For example, something similar to [this documentation](https://fastify.dev/docs/latest/Guides/Contributing/) (where there's a button on each page saying "Edit this page", linking to the source file in the repo; an idea I'd also like to incorporate for the calendar data, with links pointing to the corresponding source files).

So yes, the source documentation must absolutely reside close to the source code, in the main repo (and no wikis in GitHub, as there's no versioning, and no easy way for anyone to contribute; and not in a separate package either). Everything in the main romcal repo should remain the reference and the source of truth.

But before all of that, the number one priority is to update the documentation 😄
